src/ol_infrastructure/components/services/k8s.py

```diff
@@ -27,10 +39,28 @@ class OLApplicationK8sConfiguration(BaseModel):
     db_creds_secret_name: str
     redis_creds_secret_name: str
     static_secrets_name: str
-    application_security_group_id: str
-    application_docker_tag: str | None
+    application_security_group_id: Output[str]
+    application_security_group_name: Output[str]
+    application_image_repository: str
+    application_docker_tag: str
     vault_k8s_resource_auth_name: str
     import_nginx_config: bool
+    resource_requests: dict[str, str] = Field(
+        default={"cpu": "250m", "memory": "300Mi"}
+    )
+    resource_limits: dict[str, str] = Field(
+        default={"cpu": "500m", "memory": "600Mi"},
+    )
+    init_migrations: bool = Field(default=True)
+    init_collectstatic: bool = Field(default=True)
+
+    @field_validator("application_security_group_id")
+    def validate_sec_group_id(application_security_group_id: Output[str]):  # noqa: N805
+        return Output.from_input(application_security_group_id)
+
+    @field_validator("application_security_group_name")
+    def validate_sec_group_name(application_security_group_name: Output[str]):  # noqa: N805
+        return Output.from_input(application_security_group_name)
```

Comment:
What is going on here?

Reply:
@blarghmatey In truth I don't understand this either. I tried tracking down the apidoc for from_put because "Own your code" so I could give an explanation and no joy.